vmm/src/cpu.rs

```diff
@@ -909,6 +918,37 @@ impl CpuManager {
                                         exit_evt.write(1).unwrap();
                                         break;
                                     }
+                                    #[cfg(feature = "tdx")]
+                                    VmExit::Tdx => {
```

Comment:
What about encapsulating this inside the hypervisor crate so you don't have to leak so much into the vmm.

`hypervisor::Vcpu::get_tdx_exit_details() -> TdxExitDetails`

Reply:
And `TdxExitDetails` would be a struct holding `&mut kvm_run`? I really need a mutable reference here as I want to modify the content of the kvm_run structure.

Reply:
> You could go even further and interpret them into a `VmExit::Tdx(TdxReason)` where `TdxReason` is a Rust enum storing the actual values.
> 
> > And `TdxExitDetails` would be a struct holding `&mut kvm_run`? I really need a mutable reference here as I want to modify the content of the kvm_run structure.
> 
> Add mutators to set the values.

I tried but unfortunately to know about the exit reason I have to call `get_kvm_run()` from inside the `run()` implementation (talking about `hypervisor/src/kvm/mod.rs`), which requires a mutable reference to self but we already borrowed from `self.fd.run()` so it gets things very complicated.